compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1034,6 +1034,8 @@ impl<'a> Parser<'a> {
             })
         } else if self.eat_keyword(kw::Unsafe) {
             self.parse_block_expr(None, lo, BlockCheckMode::Unsafe(ast::UserProvided), attrs)
+        } else if self.eat_keyword(kw::Const) {
+            self.parse_const_expr(lo.to(self.token.span))
```

Comment:
There's a comment at the top of this `if`/`else` chain that says: Note: when adding new syntax here, don't forget to adjust `TokenKind::can_begin_expr()`". Did you do that?

Reply:
that comment should probably refer to `Token::can_begin_expr()` instead of `TokenKind` (and yes, it appears it was done [here](https://github.com/rust-lang/rust/pull/77124/files#diff-4792c9bf4e039cb1aba01cae59022191R166))